wombat/src/main/include/behaviour/Trigger.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2023-2024 CurtinFRC
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#pragma once
+
+#include <units/time.h>
+
+#include <functional>
+#include <string>
+
+#include "behaviour/Behaviour.h"
+
+namespace behaviour {
+/**
+ * @breif A class which binds behaviours to boolean conditions. Most commonly used through a wrapper around
+ * a controller.
+ *
+ * Trigger is an API within the Behaviour scheduling to run different Behaviours based off boolean conditions.
+ * It is instantiated with a condition that is evaluated on every Tick of the BehaviourScheduler causing the
```

Comment:
"with a [lambda] condition"

Reply:
The condition needs to be lazily evaluated so lambda doesn't need to be stated here, if we did state it it would be "a function containing a condition"